tests/network/vmi_istio.go

```diff
@@ -0,0 +1,244 @@
+				&interfaceWithExplicitPorts,
+			)
+			vmiWithDefaultPorts = newRandomCirrosVMIWithIstioProxyAndEphemeralDiskAndUserDataAndAndServiceAccountDiskAndNetworkAndInterface(
+				"default",
+				vmiDefaultPortsAppSelector,
+				"true",
+				"#!/bin/bash\necho 'hello'\n",
+				&defaultNetwork,
+				&defaultInterface,
+			)
+			vmiWithExplicitPortsService := newServiceWithAppSelector(fmt.Sprintf("%s-service", vmiWithExplicitPorts.Name), vmiWithExplicitPorts.Namespace, vmiExplicitPortsAppSelector)
+			vmiWithDefaultPortsService := newServiceWithAppSelector(fmt.Sprintf("%s-service", vmiWithDefaultPorts.Name), vmiWithDefaultPorts.Namespace, vmiDefaultPortsAppSelector)
+
+			addLabelToNamespace(vmiWithExplicitPorts.Namespace, istioInjectNamespaceLabel, "enabled")
+			for _, svc := range []corev1.Service{*vmiWithExplicitPortsService, *vmiWithDefaultPortsService} {
+				_, err = virtClient.CoreV1().Services(svc.Namespace).Create(context.Background(), &svc, metav1.CreateOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
+			}
+			for _, networkVMI := range []*v1.VirtualMachineInstance{vmiWithExplicitPorts, vmiWithDefaultPorts} {
+				_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(networkVMI)
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			// Wait for VMIs to become ready
+			vmiWithExplicitPorts = tests.WaitUntilVMIReady(vmiWithExplicitPorts, libnet.WithIPv6(console.LoginToCirros))
+			vmiWithDefaultPorts = tests.WaitUntilVMIReady(vmiWithDefaultPorts, libnet.WithIPv6(console.LoginToCirros))
+
+			for _, vmi := range []*v1.VirtualMachineInstance{vmiWithExplicitPorts, vmiWithDefaultPorts} {
+				tests.StartHTTPServer(vmi, svcDeclaredTestPort)
+				tests.StartHTTPServer(vmi, svcUndeclaredTestPort)
+			}
+		})
+
+		table.DescribeTable("should be able to reach", func(destination string, port int) {
+			var srcVMI, dstVMI *v1.VirtualMachineInstance
+			switch destination {
+			case "vmiWithExplicitPorts":
+				srcVMI = vmiWithDefaultPorts
+				dstVMI = vmiWithExplicitPorts
+			case "vmiWithDefaultPorts":
+				srcVMI = vmiWithExplicitPorts
+				dstVMI = vmiWithDefaultPorts
+			}
+			dstVMIAddress := dstVMI.Status.Interfaces[0].IP
+
+			By(fmt.Sprintf("checking that service running on port %d can be reached", port))
+			cmdCheck := fmt.Sprintf("curl %s:%d", dstVMIAddress, port)
+			err = console.SafeExpectBatch(srcVMI, []expect.Batcher{
+				&expect.BSnd{S: "\n"},
+				&expect.BExp{R: console.PromptExpression},
+				&expect.BSnd{S: cmdCheck},
+				&expect.BExp{R: console.PromptExpression},
+				&expect.BSnd{S: "echo $?\n"},
+				&expect.BExp{R: console.RetValue("0")},
+			}, 180)
+		},
+			table.Entry("declared port in VMI with explicit ports", "vmiWithExplicitPorts", svcDeclaredTestPort),
+			table.Entry("undeclared port in  VMI with explicit ports", "vmiWithExplicitPorts", svcUndeclaredTestPort),
+			table.Entry("declared port in VMI with default masquerade interface", "vmiWithDefaultPorts", svcUndeclaredTestPort),
```

Comment:
`svcDeclaredTestPort` should be passed, isn't it?
And we know this scenario is not working AFAIU, so we need to add a skip and link to an issue.

Reply:
Good catch! Copy paste errors are killing me here really.

The first commit in this PR fixes this case: `undeclared port in VMI with explicit ports`, without the iptable rule, the traffic isn't routed to the VM after it's proxied (because it's proxied from the 127.0.0.6). 

The last two cases do work in a way that the traffic does get into the VM, but it's not proxied. That is not being addressed in this PR. Ultimately though, we always want the traffic to reach the VM so IMO we should keep these cases.

Reply:
Fixed